vault_core/src/config.rs

```diff
@@ -83,15 +85,18 @@ impl Config {
     }
 
     pub fn deposit_withdrawal_fee_cap_bps(&self) -> u16 {
-        self.deposit_withdrawal_fee_cap_bps.into()
+        assert!(u16::from(self.deposit_withdrawal_fee_cap_bps) <= MAX_BPS);
```

Comment:
imo worth doing on sets and not gets. better design to prevent state from getting "corrupted" than failing post-corruption